tests/lib/rules/button-has-type.js

```diff
@@ -73,13 +76,13 @@ ruleTester.run('button-has-type', rule, {
     {
       code: '<button type={foo}/>',
       errors: [{
-        message: 'The button type attribute must be specified by a static string'
+        message: 'The button type attribute must be specified by a static string or a trivial ternary expression'
       }]
     },
     {
       code: '<button type={"foo"}/>',
       errors: [{
-        message: 'The button type attribute must be specified by a static string'
+        message: '"foo" is an invalid value for button type attribute'
```

Comment:
let's please add this test case also with single quotes, and also with backticks (it should support no-substitution-template-strings as well, and i believe that test would fail atm)

Reply:
I don't think it should work with backticks. `` `foo` `` is actually a template string not a literal. Given that there's a fixed number of possible values, it doesn't seem practical to evaluate templates here

Reply:
I'm not sure why? Plenty of rules work with "no substitution" template literals, and i think this one should too. That's a much more common use case than a "trivial ternary".